docs/pip.md

```diff
@@ -35,6 +35,33 @@ It also generates two targets for running pip-compile:
 | kwargs |  other bazel attributes passed to the "_test" rule   |  none |
 
 
+<a name="#package_annotation"></a>
```

Comment:
I think the docs here brush over some of the advanced use of this feature, or don't go into much detail about how a user might hold this. The example in the examples and mock dir is great, at a minimum this should link to it. Perhaps add an example of supplying a BUILD file for some cc_library, as that might be a common task here?

Reply:
I can do that. Can you suggest a package to use for the example? Hopefully something that doesn't require compiling something massive 😅

Reply:
Heh, I don't have a good one off the top of my head, there's normally some hidden complexity.

Reply:
Agreed. If someone can suggest one then I can try to add it to docs later. But I think the demo shows the general functionality of each annotation one could apply.